tests/migration/nodeselector.go

```diff
@@ -0,0 +1,247 @@
+
+		By("Migrating again the VM without configuring a nodeselector")
+		migration = libmigration.New(vmi.Name, vmi.Namespace)
+		libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+
+		By("Checking nodeSelector on the VMI")
+		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(vmi.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+
+		By("Checking nodeSelector on virt-launcher pod")
+		virtLauncherPod, err = libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(virtLauncherPod.Spec.NodeSelector).ToNot(HaveKeyWithValue(cleanup.TestLabelForNamespace(vmi.Namespace), "target"))
+	})
+
+	It("Should fail the migration when the nodeSelector could not be satisfied", decorators.Periodic, func() {
+		By("starting a VM on the source node")
+		vmi := libvmifact.NewFedora(
+			libnet.WithMasqueradeNetworking(),
+			libvmi.WithResourceMemory(fedoraVMSize),
+		)
+		vmi, err := virtClient.VirtualMachineInstance(testsuite.GetTestNamespace(vmi)).Create(context.Background(), vmi, metav1.CreateOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		vmi = libwait.WaitForSuccessfulVMIStart(vmi, libwait.WithTimeout(180))
+
+		sourceNodeName := vmi.Status.NodeName
+		var targetNodeName string
+
+		By("labeling a target node")
+		for _, node := range nodes.Items {
+			if node.Name != sourceNodeName {
+				targetNodeName = node.Name
+				libnode.AddLabelToNode(node.Name, cleanup.TestLabelForNamespace(vmi.Namespace), "validTarget")
+				break
+			}
+		}
+		Expect(targetNodeName).ToNot(BeEmpty(), "There should be a labeled target node")
+
+		By("Starting the migration with an unsatisfiable nodeSelector")
+		migration := libmigration.New(vmi.Name, vmi.Namespace)
+		migration.Spec.AddedNodeSelector = map[string]string{cleanup.TestLabelForNamespace(vmi.Namespace): "brokenTarget"}
+		libmigration.RunMigrationAndExpectFailure(migration, 360)
+
+		By("Checking that the VMI is still on the source node")
+		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(vmi.Status.NodeName).To(Equal(sourceNodeName))
+
+		By("Migrating again the VM without configuring a nodeSelector")
+		migration = libmigration.New(vmi.Name, vmi.Namespace)
+		libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+
+		By("Checking that the VMI is now on a different node")
+		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.GetName(), metav1.GetOptions{})
+		Expect(err).ShouldNot(HaveOccurred())
+		Expect(vmi.Status.NodeName).ToNot(Equal(sourceNodeName))
+
+	})
+
+	Context("with a node selector on the VMI", func() {
```

Comment:
I believe this is unit tested and can actually be squashed to the first test.

Reply:
done